src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.Shell/Helpers/ShellListPageHelpers.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CmdPal.Ext.Shell.Commands;
+using Microsoft.CmdPal.Extensions.Helpers;
+
+namespace Microsoft.CmdPal.Ext.Shell.Helpers;
+
+public class ShellListPageHelpers
+{
+    private static readonly CompositeFormat WoxPluginCmdCmdHasBeenExecutedTimes = System.Text.CompositeFormat.Parse(Properties.Resources.wox_plugin_cmd_cmd_has_been_executed_times);
+    private readonly SettingsManager _settings;
+
+    public ShellListPageHelpers(SettingsManager settings)
+    {
+        _settings = settings;
+    }
+
+    private ListItem GetCurrentCmd(string cmd)
+    {
+        ListItem result = new ListItem(new ExecuteItem(cmd, _settings))
+        {
+            Title = cmd,
+            Subtitle = Properties.Resources.wox_plugin_cmd_plugin_name + ": " + Properties.Resources.wox_plugin_cmd_execute_through_shell,
+            Icon = new(string.Empty),
+        };
+
+        return result;
+    }
+
+    private List<ListItem> GetHistoryCmds(string cmd, ListItem result)
+    {
+        IEnumerable<ListItem> history = _settings.Count.Where(o => o.Key.Contains(cmd, StringComparison.CurrentCultureIgnoreCase))
+            .OrderByDescending(o => o.Value)
+            .Select(m =>
+            {
+                if (m.Key == cmd)
+                {
+                    // Using CurrentCulture since this is user facing
+                    result.Subtitle = Properties.Resources.wox_plugin_cmd_plugin_name + ": " + string.Format(CultureInfo.CurrentCulture, WoxPluginCmdCmdHasBeenExecutedTimes, m.Value);
```

Comment:
After playing with this a bit: I think the `Properties.Resources.wox_plugin_cmd_plugin_name + ": "` prefix is superfluous. We're already on the shell page here. At that point, the `Shell: ` text is redundant with the page title. 

It'd make more sense on a fallback, but not here?

Reply:
I'll play around with it and report back... what you're saying makes sense. 

I haven't had luck yet implement a fallback